src/System.IO.Compression/tests/DeflateStreamTests.netstandard1.7.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public class DeflateStreamTests_NS17 : DeflateStreamTests
+    {
+
+        [Fact]
+        public static async void BeginEndReadTest()
+        {
+            byte[] buffer = new byte[32];
+            string testFilePath = gzTestFile("GZTestDocument.pdf.gz");
+            using (var readStream = await ManualSyncMemoryStream.GetStreamFromFileAsync(testFilePath, false, true))
+            using (var unzip = new DeflateStream(readStream, CompressionMode.Decompress, true))
+            {
+                IAsyncResult result = unzip.BeginRead(new byte[1], 0, 1, (ar) => { }, new object());
+                readStream.manualResetEvent.Set();
+                unzip.EndRead(result);
+            }
```

Comment:
You might consider an approach like this:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Requests/tests/FileWebRequestTest.cs#L273-L287
The base class defines some async methods which all the tests are then written in terms of.  Then the derived types just override those methods to provide implementations based on the XxAsync methods and the Begin/EndXx methods, so that all of the tests end up running with both sets of APIs.